scripts/test-cve-2016-2107.py

```diff
@@ -84,16 +110,47 @@ def main():
     else:
         run_only = None
 
+    if ciphers:
+        if not dhe:
+            # by default send minimal set of extensions, but allow user
+            # to override it
+            dhe = ciphers[0] in CipherSuite.ecdhAllSuites or \
+                    ciphers[0] in CipherSuite.dhAllSuites
+    else:
+        # any CBC will do, but the rest of code expects a 128 bit block cipher
+        # and 160 bit HMAC
+        if dhe:
+            ciphers = [CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+                       CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+                       CipherSuite.TLS_DHE_RSA_WITH_AES_128_CBC_SHA]
+        else:
+            ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+
     conversations = {}
 
     conversation = Connect(host, port)
     node = conversation
-    # any CBC will do, but the rest of code expects a 128 block cipher
-    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+    ext = {ExtensionType.renegotiation_info: None}
+    if ems:
+        ext[ExtensionType.extended_master_secret] = AutoEmptyExtension()
+    if dhe:
+        groups = [GroupName.secp256r1,
```

Comment:
Do we want to make this work in `FIPS:OSPP` and `FUTURE` by adding `secp384r1`?

Reply:
not really, as that extends to signature algorithms too, so we need to say what's good enough, and P-256 should be that